tools/attach_logs_in_jira.py

```diff
@@ -0,0 +1,158 @@
+        for i, chunk in enumerate(iter(read_chunk, b"")):
+            with open(f"{path}-part-{i}", "wb") as part_file:
+                part_file.write(chunk)
+
+
+@retry.retry(exceptions=jira.JIRAError, tries=3, logger=log)
+def _attach_logs_for_cluster(jira_client: jira.JIRA, cluster_logs_dir: Path):
+    issue = _get_or_create_jira_ticket(jira_client, JIRA_SUMMARY.format(failure_id=cluster_logs_dir.name))
+
+    for artifact in cluster_logs_dir.iterdir():
+        if artifact.is_dir():
+            if artifact.with_suffix(".tar.gz").exists():
+                log.debug("Skip archiving %s as it's already archived", artifact)
+            else:
+                artifact = _archive_dir(artifact)
+
+        if artifact.stat().st_size > JIRA_ATTACHMENT_MAX_SIZE:
+            _split_file(artifact, chunk_size=JIRA_ATTACHMENT_MAX_SIZE)
+
+    for artifact in cluster_logs_dir.iterdir():
+        if artifact.is_dir() or artifact.stat().st_size > JIRA_ATTACHMENT_MAX_SIZE:
+            log.debug("Skip %s as it had been replaced with other files", artifact)
+            continue
+
+        matching_attachments = [
+            attachment for attachment in issue.fields.attachment if artifact.name == attachment.filename
+        ]
+        if len(matching_attachments) > 0 and matching_attachments[0].size == artifact.stat().st_size:
+            log.debug("File %s already exists at %s. Skipping upload", artifact, issue)
+            continue
+
+        log.info("Attaching %s to issue %s", artifact, issue)
+        jira_client.add_attachment(issue=issue, attachment=str(artifact))
+
+
+def attach_logs_in_jira(source_dir: Path, jira_access_token: str, days_ago_threshold: Optional[int]=None):
+    jira_client = jira.JIRA(JIRA_SERVER, token_auth=jira_access_token, validate=True, max_retries=0)
+    errors = []
+
+    for cluster_logs_dir in source_dir.iterdir():
+        if days_ago_threshold is not None and days_ago(cluster_logs_dir.name) > days_ago_threshold:
+            log.debug(
+                "Ignoring old cluster dir %s as it was created more than %d days ago", cluster_logs_dir.name, days_ago
+            )
+            continue
+
+        try:
+            _attach_logs_for_cluster(jira_client, cluster_logs_dir)
+        except (jira.JIRAError, RuntimeError) as err:
+            errors.append(err)
+
+    if errors:
+        for error in errors:
+            log.error("Error while attaching logs in Jira", exc_info=error)
+
+        raise ExceptionGroup("Failed attaching logs for all issues", errors)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Create a Jira ticket for each installation attempt, and upload artifacts as Jira attachments."
+    )
+
+    parser.add_argument("--source-dir", help="Local directory with logs for all clusters")
+    parser.add_argument("-v", "--verbose", action="store_true", help="Output verbose logging")
+    parser.add_argument(
+        "--days-ago-threshold",
```

Comment:
not the best naming in the world, need to rethink about it

Reply:
`--max-days` ?

Reply:
yeah much better
thanks!